ledfx/effects/audio.py

```
@@ -620,20 +642,53 @@ def open_audio_stream(device_idx):
+        def try_open_device(dev_idx, reinit=False):
+            """
+            Attempt to open an audio device, optionally reinitializing
+            PortAudio first (clears poisoned state from WDM-KS devices).
+            Returns True on success, False on failure.
+            """
+            if reinit:
+                self.deactivate()
+                try:
+                    sd._terminate()
+                    sd._initialize()
+                except Exception as reinit_err:
```

Ruff (0.15.9)

[warning] Do not catch blind exception: `Exception`

(BLE001)